scripts/o3de/tests/test_utils.py

```diff
@@ -37,3 +39,16 @@ def test_validate_identifier(value, expected_result):
 def test_validate_uuid4(value, expected_result):
     result = utils.validate_uuid4(value)
     assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "in_list, out_list", [
+        pytest.param(['A', 'B', 'C'], ['A', 'B', 'C']),
+        pytest.param(['A', 'B', 'C', 'A', 'C'], ['A', 'B', 'C']),
+        pytest.param(['A', {'name': 'A', 'optional': True}], [{'name': 'A', 'optional': True}]),
+        pytest.param([{'name': 'A', 'optional': True}, 'A'], ['A']),
```

Comment:
Can you add a test parameters for when the optional field is missing or for when the name field is missing as well?

```suggestion
        pytest.param(['A', 'B', 'C'], ['A', 'B', 'C']),
        pytest.param(['A', 'B', 'C', 'A', 'C'], ['A', 'B', 'C']),
        pytest.param(['A', {'name': 'A', 'optional': True}], [{'name': 'A', 'optional': True}]),
        pytest.param([{'name': 'A', 'optional': True}, 'A'], ['A']),
        pytest.param([{'name': 'A'}], [{'name': 'A'}]),
        pytest.param([{'optional': False}], []),
```